modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/GlobalSnapshotStateChannelEventsProcessor.scala

```diff
@@ -17,45 +17,49 @@ import scala.collection.immutable.SortedMap
 import org.tessellation.currency.schema.currency._
 import org.tessellation.ext.cats.syntax.validated._
 import org.tessellation.json.JsonBrotliBinarySerializer
-import org.tessellation.node.shared.domain.statechannel.StateChannelValidator
+import org.tessellation.node.shared.domain.statechannel.StateChannelAcceptanceResult.CurrencySnapshotWithState
+import org.tessellation.node.shared.domain.statechannel.{FeeCalculator, StateChannelAcceptanceResult, StateChannelValidator}
+import org.tessellation.schema.ID.Id
 import org.tessellation.schema.address.Address
-import org.tessellation.schema.currencyMessage.fetchStakingBalance
+import org.tessellation.schema.balance.Balance
+import org.tessellation.schema.currencyMessage.{MessageType, fetchStakingBalance}
 import org.tessellation.schema.{GlobalSnapshotInfo, SnapshotOrdinal}
 import org.tessellation.security.Hasher
+import org.tessellation.security.hash.Hash
+import org.tessellation.security.hex.Hex
 import org.tessellation.security.signature.Signed
+import org.tessellation.security.signature.signature.{Signature, SignatureProof}
 import org.tessellation.statechannel.{StateChannelOutput, StateChannelSnapshotBinary, StateChannelValidationType}
 
 import eu.timepit.refined.auto._
+import io.circe.Decoder
 import org.typelevel.log4cats.slf4j.Slf4jLogger
 
 trait GlobalSnapshotStateChannelEventsProcessor[F[_]] {
-  type CurrencySnapshotWithState = Either[Signed[CurrencySnapshot], (Signed[CurrencyIncrementalSnapshot], CurrencySnapshotInfo)]
 
   def process(
     snapshotOrdinal: SnapshotOrdinal,
     lastGlobalSnapshotInfo: GlobalSnapshotInfo,
     events: List[StateChannelOutput],
     validationType: StateChannelValidationType
-  )(implicit hasher: Hasher[F]): F[
-    (
-      SortedMap[Address, NonEmptyList[Signed[StateChannelSnapshotBinary]]],
-      SortedMap[Address, CurrencySnapshotWithState],
-      Set[StateChannelOutput]
-    )
-  ]
+  )(implicit hasher: Hasher[F]): F[StateChannelAcceptanceResult]
 
   def processCurrencySnapshots(
+    snapshotOrdinal: SnapshotOrdinal,
     lastGlobalSnapshotInfo: GlobalSnapshotInfo,
     events: SortedMap[Address, NonEmptyList[Signed[StateChannelSnapshotBinary]]]
-  )(implicit hasher: Hasher[F]): F[SortedMap[Address, NonEmptyList[CurrencySnapshotWithState]]]
+  )(
+    implicit hasher: Hasher[F]
+  ): F[(SortedMap[Address, (NonEmptyList[(Signed[StateChannelSnapshotBinary], Option[CurrencySnapshotWithState])], Map[Address, Balance])])]
```

Comment:
(unnecessary outer parentheses)
The value of the map is a complicated tuple. I feel like it deserves to be defined as a case class. It will help make the code a lot easier to follow.

Reply:
(I forgot to mention that it occurs 5 times in the file)

Reply:
Introduced type descriptions. Didn't add case classes to not introduce boilerplate.